src/analysis/normed_space/multilinear.lean

```diff
@@ -1026,7 +1026,23 @@ def continuous_multilinear_curry_right_equiv :
   end,
   .. continuous_multilinear_curry_right_equiv_aux 𝕜 E E₂ }
 
-variables {𝕜 G E E₂}
+variables (n G)
+
+/-- The space of continuous multilinear maps on `Π(i : fin (n+1)), G` is canonically isomorphic to
+the space of continuous multilinear maps on `Π(i : fin n), G` with values in the space
+of continuous linear maps on `G`, by separating the last variable. We register this
+isomorphism as a continuous linear equiv in `continuous_multilinear_curry_right_equiv' 𝕜 n G E₂`.
+For a version allowing dependent types, see `continuous_multilinear_curry_right_equiv`. When there
+are no dependent types, use the primed version as it helps Lean a lot for unification.
+
+The direct and inverse maps are given by `f.uncurry_right` and `f.curry_right`. Use these
+unless you need the full framework of continuous linear equivs. -/
+def continuous_multilinear_curry_right_equiv' :
+  (continuous_multilinear_map 𝕜 (λ(i : fin n), G) (G →L[𝕜] E₂)) ≃L[𝕜]
+  (continuous_multilinear_map 𝕜 (λ(i : fin n.succ), G) E₂) :=
+continuous_multilinear_curry_right_equiv 𝕜 (λ (i : fin n.succ), G) E₂
+
```

Comment:
Should the `simp` lemmas below have copies for the primed version?

Reply:
Done